torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3413,28 +3416,127 @@ def _make_data_dependent_error(self, expr, unhinted_expr):
             # problem
         )
 
-    def _set_replacement(self, a: "sympy.Symbol", expr: "sympy.Expr") -> None:
+    def _set_replacement(self, a: "sympy.Symbol", tgt: "sympy.Expr", msg: str, *, force_size_like: bool = False) -> None:
         """
         Adds or updates a replacement for a symbol.
-        Use this instead of `self.replacements[a] = expr`.
+        Use this instead of `self.replacements[a] = tgt`.
         """
-        if config.print_specializations and isinstance(expr, (sympy.Integer, sympy.Float)):
+
+        # Precondition: a == tgt
+
+        # Handles NT symbolic variables...
+        tgt_bound = None
+        if a in self.var_to_range:
+            src_bound = self.var_to_range[a]
+
+            # If you have x in [2, maxint], then 2*x in [4, 2*maxint].
+            # But we don't really care that the max bound says we can
+            # go beyond the maximum integer size, because we aren't
+            # using bigints anyway.  Arguably, ValueRanges should know
+            # to do this truncation automaticaly (to avoid doing
+            # bigint compute in range analysis), but right now it doesn't
+            # so we need to get rid of some unnecessary precision.
+            int_range = ValueRanges(-sys.maxsize - 1, sys.maxsize - 1)
```

Comment:
Here we use `sys.maxsize-1` as an upper bound and in `_default_unspecified_value_range` we use `sys.maxsize`. Is that expected?

Reply:
Yes, this is the sys.maxsize carveout for size-like

```
    # See: Note - On 0/1 specialization
    # NB: sys.maxsize is NOT allowed for sizes, because we use MAX_INT
    # as a sentinel sometimes.  Your sizevar isn't going to be
    # anywhere near the max 64-bit integer anyway.
    def _default_value_range(self) -> ValueRanges:
        lower = 2 if self.specialize_zero_one else 0
        return ValueRanges(lower, sys.maxsize - 1
```